invenio_records_rest/__init__.py

```diff
@@ -22,7 +22,634 @@
+serializers for the multiple results returned by a search query, and the
+single result returned when querying for specific records.
+This can be configured in the following part of the records rest endpoints:
+
+>>> RECORDS_REST_ENDPOINTS = dict(
+...    recid=dict(
+...       record_serializers=dict(
+...          application/json=('invenio_records_rest.serializers'
+...                            ':json_v1_response'),
+...       ),
+...       search_serializers=dict(
+...          application/json=('invenio_records_rest.serializers'
+...                            ':json_v1_search'),
+...       )
+...    )
+... )
+
+With the usage of MIME types it is also possible to specify the
+version of a given serialization, as serialization schemas can have
+newer versions over time. More specifically, if we have created two
+serializers, one for DataCite's version 3.1 metadata schema and one
+for version 4.1, we can specify which one we require in the
+Content-Type http header field. For example, if we have
+set in the configuration:
+
+>>> record_serializer={
+...    'application/datacite.v3.1+xml': ('invenio_records_rest.serializers'
+...                                      'datacite_response_v31'),
+...    'application/datacite.v4.1+xml': ('invenio_records_rest.serializers'
+...                                      'datacite_response_v41'),
+... }
+
+
+Then we can send an http request with the http headers containing
+``Content-Type:application/datacite+xml`` or
+``Content-Type:application/datacite-v41+xml`` and receive the output
+in the different formats.
+
+Workflow
+~~~~~~~~
+The serializers typically will follow a certain number of steps
+to process records.
+For example, for records returned as results to a search query the format
+is dictated from Elasticsearch, while for requests for specific records
+the format comes from the database. The output has to be consistent so a
+first step is to create the different methods to syncronize the data which
+will be returned. The abstract classes with the boilerplate for this workflow
+are in `invenio_records_rest.serializers.base`.
+
+Additionally, we can define more steps to anonymize the data, if sensitive
+information need to be excluded or enrich it, by adding extra fields.
+
+Citation formatting
+~~~~~~~~~~~~~~~~~~~
+A common application of serialization is to output a citation from the
+record content. For this task the ``CiteprocSerializer``, which in turn is
+based on the `citeproc-py <https://github.com/brechtm/citeproc-py>`_
+module, is used. To produce the citation text, a schema is required which
+defines what fields have to be present. For example the following schemas
+will create a citation with the title, abstract and authors of a given record:
+
+
+>>> from marshmallow import Schema, fields
+>>> class AuthorSchema(Schema):
+...    family = fields.Method('get_family_name')
+...
+...    def get_family_name(self, obj):
+...        return obj['name']
+
```

Comment:
Add a `>>>` here since the code block render as disconnected:

![records-rest](https://user-images.githubusercontent.com/1814661/37357328-6ff0ac22-26e8-11e8-94fa-d70778b691fa.png)